hs_collection_resource/models.py

```diff
@@ -0,0 +1,137 @@
+        # no file types are supported
+        return ()
+
+    @classmethod
+    def can_have_multiple_files(cls):
+        # resource can't have any files
+        return False
+
+    @property
+    def metadata(self):
+        md = CollectionMetaData()
+        return self._get_metadata(md)
+
+processor_for(CollectionResource)(resource_processor)
+
+class CollectionItems(AbstractMetaDataElement):
+    term = 'CollectionItems'
+    collection_items = models.ManyToManyField(BaseResource, null=True, blank=True)
+
+    def get_collection_items_str(self):
+        return ','.join([res.title+":" + res.resource_type for res in self.collection_items.all()])
+
+    @classmethod
+    def create(cls, **kwargs):
+        if 'collection_items' in kwargs:
+            metadata_obj = kwargs['content_object']
+            new_meta_instance = CollectionItems.objects.create(content_object=metadata_obj)
+            for res_id in kwargs['collection_items']:
+                res = BaseResource.objects.filter(short_id__iexact=res_id)
+                if res.exists():
+                    new_meta_instance.collection_items.add(res[0])
+            return new_meta_instance
+        else:
+            raise ObjectDoesNotExist("No collection_items parameter was found in the **kwargs list")
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        meta_instance = CollectionItems.objects.get(id=element_id)
+        if meta_instance:
+            if 'collection_items' in kwargs:
+                meta_instance.collection_items.clear()
+                for res_id in kwargs['collection_items']:
+                    qs = BaseResource.objects.filter(short_id__iexact=res_id)
+                    if qs.exists():
+                        meta_instance.collection_items.add(qs[0])
+                meta_instance.save()
+                if meta_instance.collection_items.all().count() == 0:
+                    meta_instance.delete()
+            else:
+                raise ObjectDoesNotExist("No collection_items parameter was found in the **kwargs list")
+        else:
+            raise ObjectDoesNotExist("No CollectionItems object was found with the provided id: %s" % kwargs['id'])
+
+    @classmethod
+    def remove(cls, element_id):
+        raise ValidationError("CollectionItems element can't be deleted.")
+
+
+class CollectionMetaData(CoreMetaData):
+    # collection_items = generic.GenericRelation(CollectionItems)
+    collection_items = GenericRelation(CollectionItems)
+
+    @classmethod
+    def get_supported_element_names(cls):
+        elements = super(CollectionMetaData, cls).get_supported_element_names()
+        elements.append('CollectionItems')
+        return elements
+
+    def has_all_required_elements(self):
+        if self.get_required_missing_elements():
+            return False
+        else:
+            # check if no member resource exists
+            if self.collection_items.first().collection_items.all().count() == 0:
```

Comment:
The naming of the classes makes it difficult to read this statement. Refer to where I have suggested changing class names.